File: 1_intro/2_edge_of_the_ocean/4_adjacentElementsProduct/adjacentElementsProduct.js
function adjacentElementsProduct(inputArray) {
    var biggest = inputArray[0] * inputArray[1],
        second = inputArray[1] * inputArray[2],
        len = inputArray.length;
    if (biggest < second) {
        biggest = inputArray[1] * inputArray[2];
        second = inputArray[0] * inputArray[1];
    }
    for (i = 2; i < len; i++) {
        var check = inputArray[i] * inputArray[i+1];
        if (check > biggest){
            second = biggest;
            biggest = check;
        }
        else if (check > second) {
            second = check;
        }
    }
    return biggest;
}

File: 1_intro/2_edge_of_the_ocean/4_adjacentElementsProduct/adjacentElementsProduct.lua
function adjacentElementsProduct(inputArray)
  len = table.getn(inputArray)
  biggest = inputArray[1] * inputArray[2]
  if len == 2 then 
    return biggest 
  end
  second = inputArray[2] * inputArray[3]  
  if biggest < second then
    biggest = inputArray[2] * inputArray[3]
    second = inputArray[1] * inputArray[2]
  end
  for i=3,len-1 do
    check = inputArray[i] * inputArray[i+1]
    if check > biggest then
      second = biggest
      biggest = check
    elseif check > second then
      second = check
    end
  end
  return biggest
end